bin/select_gfa

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+
+import desimodel.focalplane
+import desitarget.io
+import fitsio
+import numpy as np
+import desimodel.io
+import desimodel.footprint
+import matplotlib.pyplot as plt
+import glob
+import time
+import os.path
+import argparse
+
+def near_tile(data, tilera, tiledec, window_ra=4.0, window_dec=4.0):
+    delta_RA = data['RA'] - tilera
+    delta_dec = data['DEC'] - tiledec
+    jj = np.fabs(delta_RA) < window_ra
+    jj = jj | ((delta_RA + 360.0) < window_ra)
+    jj = jj | ((360.0 - delta_RA) < window_ra)
+    jj = jj & (np.fabs(delta_dec)<window_dec)
+    return jj
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('-o', '--output_dir', help='Path to write the outputs', type=str)
+parser.add_argument('-s', '--input_sweep', help='Path to the sweep files', type=str)
+parser.add_argument('-f', '--input_fiberassign', help='Path to fiberassign output files', type=str)
+parser.add_argument('-t', '--input_tiles', help='Input file with the tileid gfa files to be computed', type=str)
+
+args = parser.parse_args()
+
+# list tilefiles already available
+if args.input_fiberassign:
+    fiberassign_tilefiles = glob.glob(os.path.join(args.input_fiberassign,"tile*.fits"))
+    print(len(fiberassign_tilefiles), 'fiberassign tile files')
+else:
+    fiberassign_tilefiles = []
+
+if not os.path.isdir(args.output_dir):
+    os.makedirs(args.output_dir, exist_ok=True)
+
+# list sweep files to be used
+sweep_files = desitarget.io.list_sweepfiles(args.input_sweep)
+n_sweep = len(sweep_files)
+print(len(sweep_files), 'sweep files')
+
+# list of tiles to be computed
+desi_tiles = desimodel.io.load_tiles()
+if args.input_tiles is not None:
+    gfa_tiles = np.int_(np.loadtxt(args.input_tiles))
+    print(len(gfa_tiles), 'tiles to process')
+
+    # Check that all of them are known DESI tiles
+    unknown_tiles = ~np.in1d(gfa_tiles, desi_tiles['TILEID'])
+    if np.any(unknown_tiles):
+        nbad = np.count_nonzero(unknown_tiles)
+        print('{} unknown tiles: {}'.format(nbad, gfa_tiles[unknown_tiles]))
+
+    # Trim DESI tiles list to the requested subset
+    ii = np.in1d(desi_tiles['TILEID'], gfa_tiles)
+    desi_tiles = desi_tiles[ii]
+
+#load all sweep data
+sweep_data = []
+# n_sweep = 10
+for i in range(n_sweep):
+    sweep_file = sweep_files[i]
+    data = fitsio.read(sweep_file, columns=['RA', 'DEC', 'FLUX_R'])
+
+    #- Keep just mag>18
+    rfluxlim = 10**(0.4*(22.5-18))
+    ii = data['FLUX_R'] > rfluxlim
+    data = data[ii]
```

Comment:
This cut can be done on-the-fly using fitsio's `where` method, without having to read in the full sweep, which should save time and tons of memory.

Reply:
Memory is not a problem. We are using only 1GB of RAM. Time is not too bad. It takes 2 minutes to read all the sweeps.